packages/apollo-federation/src/service/__tests__/buildFederatedSchema.test.ts

```diff
@@ -10,40 +17,88 @@ const EMPTY_DOCUMENT = {
   definitions: [],
 };
 
-describe('buildFederatedSchema', () => {
-  it(`should mark a type with a key field as an entity`, () => {
-    const schema = buildFederatedSchema(gql`
-      type Product @key(fields: "upc") {
-        upc: String!
-        name: String
-        price: Int
-      }
-    `);
+const testFederatedSchema = (
+  name: string,
+  createSchema: () => GraphQLSchema,
+  runTest: (schema: GraphQLSchema) => Promise<void>,
+) => {
+  const schema = createSchema();
```

Comment:
Why pass the `schema`'s as thunks if you're invoking them only this single time? At first I thought you were passing as thunk to avoid issues with modifing the schema in place, but it looks like you're passing the same item to both tests.

Reply:
True, I forgot why I went this route. I was rushing it by being clever and reusing existing tests instead of rewriting them.. which I suppose is not a terrible thing. :) I can change it though.